extensions/eclipse-che-theia-user-preferences/src/browser/che-frontend-application.ts

```diff
@@ -0,0 +1,70 @@
+/**********************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+import {
+  ApplicationShell,
+  FrontendApplication,
+  FrontendApplicationContribution,
+  KeybindingRegistry,
+  ShellLayoutRestorer,
+} from '@theia/core/lib/browser';
+import { CommandRegistry, ContributionProvider, MenuModelRegistry } from '@theia/core';
+import { inject, injectable, named } from 'inversify';
+
+import { CheShellLayoutRestorer } from './che-shell-layout-restorer';
+import { FrontendApplicationStateService } from '@theia/core/lib/browser/frontend-application-state';
+import { StorageServicePreferences } from './che-storage-preferences';
+
+@injectable()
+export class CheFrontendApplication extends FrontendApplication {
+  protected layoutPersistenceTimer: number | undefined;
+
+  constructor(
+    @inject(CommandRegistry) protected readonly commands: CommandRegistry,
+    @inject(MenuModelRegistry) protected readonly menus: MenuModelRegistry,
+    @inject(KeybindingRegistry) protected readonly keybindings: KeybindingRegistry,
+    @inject(CheShellLayoutRestorer) protected readonly layoutRestorer: ShellLayoutRestorer,
+    @inject(ContributionProvider)
+    @named(FrontendApplicationContribution)
+    protected readonly contributions: ContributionProvider<FrontendApplicationContribution>,
+    @inject(ApplicationShell) protected readonly _shell: ApplicationShell,
+    @inject(FrontendApplicationStateService) protected readonly stateService: FrontendApplicationStateService,
+    @inject(StorageServicePreferences) protected readonly storagePreferences: StorageServicePreferences
+  ) {
+    super(commands, menus, keybindings, layoutRestorer, contributions, _shell, stateService);
+  }
+
+  protected registerEventListeners() {
+    super.registerEventListeners();
+
+    window.addEventListener('resize', () => this.scheduleLayoutPersistence());
+    document.addEventListener('keydown', () => this.scheduleLayoutPersistence());
+    document.addEventListener('mousedown', () => this.scheduleLayoutPersistence());
+    document.addEventListener('mousemove', () => this.scheduleLayoutPersistence());
+  }
+
+  protected scheduleLayoutPersistence() {
+    if (this.layoutPersistenceTimer) {
```

Comment:
looks like it's always `true` except the first method call

Reply:
It is true. Answered in https://github.com/eclipse/che-theia/pull/970#discussion_r563931483

Reply:
so we can remove it at all? 
:-)

Reply:
Removed